lib/assessment.rb

```diff
@@ -5,35 +5,52 @@
 # Create a class called Dog. It should have, as attributes, name, breed, and age.
 # Make all of those attributes read-only.
 
-##
-# your code here
-##
+class Dog
+  attr_reader :name, :breed, :age
+
+  def initialize(name, breed, age)
+    @name = name
+    @breed = breed
+    @age = age
+  end
+end
 
 # Question 2
 # Create a new Dog object.
 
-Answer.dog = nil # replace nil with your code
+Dog.new("Veto", "japanese chin", 8)
 
 # Question 3
 # Create a Person class, with attributes name, age, and location.
 # It should set all three attributes when you create a new Person.
 # Name and location should be writeable, but age should be read-only.
 
-##
-# your code here
-##
+class Person
+  attr_writer :name, :location
```

Comment:
Although this will make `name` and `location` writeable, it won't make them readable. This is because `attr_writer` only creates 'getter' methods, _not 'setter' ones_. That's a really bizarre, unusual use case, so `attr_writer` is almost never used; instead, people use `attr_accessor`. That's really the right tool here too.